devicetokens/devicetokens.go

```diff
@@ -0,0 +1,84 @@
+package devicetokens
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/tidepool-org/platform/structure"
+)
+
+type Document struct {
+	// UserID of the user that owns the DeviceToken.
+	UserID string `json:"userId" bson:"userId"`
+	// TokenID is string that uniquely identifies the DeviceToken.
+	//
+	// It's useful for generating unique indexes.
+	TokenID string `json:"tokenId" bson:"tokenId"`
```

Comment:
I'll change it to TokenKey, but I think that's a less descriptive name.

May I ask how it's misleading? What did you think it was? Was the documentation not clear? I came to this name because ID is a popular suffix for values used in a database indexes (as this one is).

If you think of IDs as only describing UUIDs found in Tidepool endpoints, I find that a too restrictive concept for such a widespread and useful term as "ID", but my brain has made enough weird associations that I can't point fingers. :)

Reply:
Okay, your code.